soseki/biencoder/modeling.py

```diff
@@ -2,57 +2,172 @@
 import random
 from argparse import ArgumentParser, Namespace
 from collections import defaultdict
-from typing import Any, Callable, Dict, List, Optional, Tuple
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple
 
 import torch
 import torch.distributed as dist
 import torch.nn as nn
 import torch.nn.functional as F
 from torch import Tensor
 from torch.optim import AdamW, Optimizer
-from torch.utils.data import Dataset, DataLoader, DistributedSampler
+from torch.utils.data import Dataset, DataLoader, DistributedSampler, Sampler
 from tqdm import tqdm
 from pytorch_lightning.core import LightningModule
-from pytorch_lightning.utilities.distributed import distributed_available, rank_zero_info
+from pytorch_lightning.utilities.distributed import distributed_available
+from pytorch_lightning.utilities.rank_zero import rank_zero_info
```

Comment:
`rank_zero_info` is moved to `pytorch_lightning.utilities.rank_zero` in the latest version of PyTorch Lightning.